src/assisted_installer_controller/operator_handler.go

```diff
@@ -48,7 +49,22 @@ func (c *controller) isOperatorAvailable(handler OperatorHandler) bool {
 			return false
 		}
 
-		err = c.ic.UpdateClusterOperator(context.TODO(), c.ClusterID, operatorName, operatorVersion, operatorStatus, operatorMessage)
+		// If the operator is available but has setup jobs then we don't want to mark it as available here
+		// because if we do that for all operators then the service will consider that the cluster is ready
+		// and that will in turn stop the controller without first checking if the setup jobs have finished.
+		// So instead we mark the operator as progressing, and will mark it as available later, when we check
+		// the setup jobs.
+		var reportedStatus models.OperatorStatus
```

Comment:
I believe all setup jobs logic should be here, if operator has setup job don't set it as available but i don't think we should split the logic